openhands/controller/agent_controller.py

```diff
@@ -323,21 +346,7 @@ async def _handle_message_action(self, action: MessageAction) -> None:
             if self.get_agent_state() != AgentState.RUNNING:
                 await self.set_agent_state_to(AgentState.RUNNING)
         elif action.source == EventSource.AGENT and action.wait_for_response:
-            # The agent has finished a task and wants to start waiting for a user response.
-            if self.state.replay_phase == ReplayDebuggingPhase.Analysis:
-                # NOTE: To simplify things for the LLM, we ask it to simply check in
-                #       with the user, after analysis.
-                # NOTE: The action was sent from `code_act_agent/function_calling.py` → `response_to_actions`
-                # TODO: Add dedicated tools for managing the analysis state machine.
-
-                # Tell the agent to stop analyzing and start editing:
-                self.state.replay_phase = ReplayDebuggingPhase.Edit
-                self.event_stream.add_event(
-                    MessageAction(content='Implement the changes.'),
-                    EventSource.USER,
-                )
-            else:
-                await self.set_agent_state_to(AgentState.AWAITING_USER_INPUT)
+            await self.set_agent_state_to(AgentState.AWAITING_USER_INPUT)
```

Comment:
NOTE: This was the old state machine. I used the agent's `AWAITING_USER_INPUT` state for it. Now we have proper state transitions.